app/controllers/users_controller.rb

```diff
@@ -0,0 +1,57 @@
+class UsersController < ApplicationController
+
+  get '/signup' do
+    if logged_in?
+      redirect '/tweets'
+    else
+      erb :"users/create_user"
+    end
+  end
+
+  get '/users/:slug' do
+    @user = User.find_by_slug(params[:slug])
+    erb :"/users/show"
+  end
+
+  post '/signup' do
+    # binding.pry
+    @user = User.new(params)
+    if params[:username].empty? || params[:email].empty? || params[:password].empty?
```

Comment:
I think you could use [ActiveRecord validations](https://learn.co/tracks/full-stack-web-development/sinatra/activerecord/user-authentication-in-sinatra) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.